openedx/core/djangoapps/public_api/courses/tests/test_course_import_export.py

```diff
@@ -0,0 +1,407 @@
+        """
+        result = self.client.get(self.url, HTTP_ACCEPT='application/json')
+
+        self.assertListEqual(
+            json.loads(result.content)['courses'],
+            [unicode(self.course.location.course_key)]
+        )
+
+    def test_multiple_courses(self):
+        """
+        Test that multiple courses are properly listed by the view.
+        """
+        CourseFactory.create(
+            org="MITx",
+            course="FabulousCourse",
+            run="Spring_2015"
+        )
+        result = self.client.get(self.url, HTTP_ACCEPT='application/json')
+        result_courses = json.loads(result.content)['courses']
+
+        self.assertIn(unicode(self.course.location.course_key), result_courses)
+        self.assertIn(u'MITx/FabulousCourse/Spring_2015', result_courses)
+
+    def test_session_authentication(self):
+        """
+        Verify that session authenticated users can make requests.
+        """
+        client = Client()
+        client.login(username=self.user.username, password=self.user_password)
+        result = client.get(self.url, HTTP_ACCEPT='application/json')
+
+        self.assertListEqual(
+            json.loads(result.content)['courses'],
+            [unicode(self.course.location.course_key)]
+        )
+
+
+@override_settings(CONTENTSTORE=TEST_DATA_CONTENTSTORE)
+class ImportTestCase(CourseTestCase):
+    """
+    Unit tests for importing a course
+    """
+    def setUp(self):
+        super(ImportTestCase, self).setUp()
+        self.url = course_url('course_import_export_handler', self.course)
+        self.content_dir = path(tempfile.mkdtemp())
+
+        # Create tar test files -----------------------------------------------
+        # OK course:
+        good_dir = tempfile.mkdtemp(dir=self.content_dir)
+        # test course being deeper down than top of tar file
+        embedded_dir = os.path.join(good_dir, "grandparent", "parent")
+        os.makedirs(os.path.join(embedded_dir, "course"))
+        with open(os.path.join(embedded_dir, "course.xml"), "w+") as f:
+            f.write('<course url_name="2013_Spring" org="EDx" course="0.00x"/>')
+
+        with open(
+                os.path.join(embedded_dir, "course", "2013_Spring.xml"),
+                "w+"
+        ) as f:
+            f.write('<course></course>')
+
+        self.good_tar = os.path.join(self.content_dir, "good.tar.gz")
+        with tarfile.open(self.good_tar, "w:gz") as gtar:
+            gtar.add(good_dir)
+
+        # Bad course (no 'course.xml' file):
+        bad_dir = tempfile.mkdtemp(dir=self.content_dir)
+        path.joinpath(bad_dir, "bad.xml").touch()
+        self.bad_tar = os.path.join(self.content_dir, "bad.tar.gz")
+        with tarfile.open(self.bad_tar, "w:gz") as btar:
+            btar.add(bad_dir)
+
+        self.unsafe_common_dir = path(tempfile.mkdtemp(dir=self.content_dir))
+
+    def tearDown(self):
+        shutil.rmtree(self.content_dir)
```

Comment:
In this case, even better, use `openedx.core.lib.tempdir.mkdtemp_clean` instead of `tempfile.mkdtemp`, then you don't need a teardown or cleanup at all.